symengine/series.h

```diff
@@ -187,7 +188,7 @@ class SeriesBase : public SeriesCoeffInterface
                                      unsigned int prec)
     {
         if (s == 0)
-            throw std::runtime_error("Series::series_invert: division by zero");
+            throw DivisionByZero();
```

Comment:
The question is how much information we want to send to the wrappers. Generally, when division by zero occurs, either it's trivial (like if the user writes `x/0`), or it's inside some algorithm in SymEngine, in which case that algorithm needs to be fixed. We can print the nice full stacktrace (with exception string) if an exception is caught in the C wrappers in Debug mode. So the user just needs to turn on Debug mode to see what is going on. In Release mode, perhaps just a simple integer is all we need to return, i.e. as you already do in the PR?